analyzer/src/main/kotlin/curation/FilePackageCurationProvider.kt

```diff
@@ -28,11 +29,27 @@ import org.ossreviewtoolkit.model.PackageCuration
 import org.ossreviewtoolkit.model.readValueOrDefault
 
 /**
- * A [PackageCurationProvider] that loads [PackageCuration]s from a single file. Supports all file formats specified
- * in [FileFormat].
+ * A [PackageCurationProvider] that loads [PackageCuration]s from all [curationFiles]. Supports all file formats
+ * specified in [FileFormat].
  */
-class FilePackageCurationProvider(curationFile: File) : PackageCurationProvider {
-    internal val packageCurations by lazy { curationFile.readValueOrDefault(emptyList<PackageCuration>()) }
+class FilePackageCurationProvider(curationFiles: Collection<File>) : PackageCurationProvider {
+    constructor(curationFile: File) : this(listOf(curationFile))
+
+    @Suppress("TooGenericExceptionCaught")
+    internal val packageCurations by lazy {
+        curationFiles.map { curationFile ->
+            try {
+                curationFile.readValueOrDefault(emptyList<PackageCuration>())
+            } catch (e: Exception) {
```

Comment:
Why is this not just `IOException` like in `SimplePackageConfigurationProvider`? If it was, we could also just rethrow it (again see `SimplePackageConfigurationProvider`) instead of coming up with our own exception.

Reply:
See: https://github.com/oss-review-toolkit/ort/pull/4131#discussion_r648073792

Reply:
Sorry, that conversion was resolved (and thus hidden) too early for me 😀

Reply:
But sorry, one more question: Do we really stop parsing for all remaining files once an exception occurred? In my upcoming [multi-location-package-config](https://github.com/oss-review-toolkit/ort/tree/multi-location-package-config) branch I'm using [this code](https://github.com/oss-review-toolkit/ort/blob/75f93cdc65c533d63449231b0139a785734dfd17/model/src/main/kotlin/utils/SimplePackageConfigurationProvider.kt#L58-L66) to only log warnings about those files, but continue parsing the other files.

Reply:
I agree. Changed it to logging the warning instead of throwing an error.